Documentation: Roles and Permissions and Groups (#9005)
opf/openproject: docs/system-admin-guide/users-permissions/roles-permissions/README.md

---
# Roles and permissions

A **role** is a set of **permissions** that can be assigned to any project member. Multiple roles can be assigned to the same project member.

> When creating a role, the "Global role" field can be ticked, making it a **Global role** that can be assigned to a [users details](../users/#manage-user-settings) and applied across all Projects.

Reviewer: ```suggestion
When creating a role, the "Global role" field can be ticked, making it a **Global role** that can be assigned to a [users details](../users/#manage-user-settings) and applied across all projects. 
```

Author: Accepted

## Permissions

> The permissions are pre-defined in the system, and cannot be changed. They define what actions a role can do. If a user has more than one role, including global and project roles, a permission is granted if it exists on any of those roles.

Reviewer: ```suggestion
The permissions are pre-defined in the system, and cannot be changed. They define what actions a role can carry out. If a user has more than one role, including global and project roles, a permission is granted if it is assigned to any of those roles.
```
not sure to be honest, what do you think?

Author: Accepted

All permissions are shown by OpenProject module in the [create a new role](#create-a-new-role) page.

### Project Modules

> If a [project module](../../user-guide/projects/project-settings/modules/) is not enabled it is not shown in the project menus whether the user has permission for that module or not.

Reviewer: Good note!
```suggestion
If a [project module](../../../user-guide/projects/project-settings/modules/) is not enabled for a specific project it is not shown in that project's menu, whether the user has permission for that module or not.
```

Author: Accepted

> ### Permissions report

Reviewer: I would suggest to move this chapter at the end/bottom, because this way the reader would be introduced to the roles list and the overview of permissions for a specific role first (which are referenced in the "permissions report" chapter.

Author: Accepted

> On the Roles list page is a link to the **Permissions report**. This shows a grid of existing roles (horizontally) against permissions (vertically); the intersections are ticked if the role has the permission.

Reviewer: That's a very helpful note for admins, I think!

one small change:
```suggestion
On the bottom of the roles list page there is a link to the **Permissions report**. This shows a grid of existing roles (horizontally) against permissions (vertically); the intersections are ticked if the role has the permission.
```

Author: Accepted

A "Check/uncheck all" tick is shown on each role or permission to allow bulk change. Be careful, this cannot be undone. If you make a mistake do not save the report.

## Create a new role

> To create a new role, navigate to the administration and select -> Users & permissions -> Roles and permissions from the menu on the left.

Reviewer: Please mark navigation paths this way:
```suggestion
To create a new role, navigate to the administration and select *-> Users & permissions -> Roles and permissions* from the menu on the left.
```

Author: Accepted

You will see the list of all the roles that have been created so far.

![create roles](image-20200211142134472.png)

> After clicking the green **+ New Role** button a form will shown to define the role.

Reviewer: ```suggestion
After clicking the green **+ Role** button a form will be shown to define the role and its permissions.
```
Great that you caught the mistake (wrong fromatting of "+ New Role"!

Author: Accepted

> ![Sys-admin-create-new-role](Sys-admin-create-new-role.png)

Reviewer: please put this picture after the description (i.e. after "...at the bottom of the page")

Author: Accepted

Complete the following as required:

> * **Role name** - must be entered and be a new name.

Reviewer: ```suggestion
1. **Role name** - must be entered and be a new name.
```

Author: Accepted

> * **Global Role** - this role applies to all projects, and is assigned in the [user details](../users/#manage-user-settings). Once saved this field is not shown if it was not ticked, so it cannot be changed.

Reviewer: ```suggestion
2. **Global Role** - this role applies to all projects and can only be assigned in the [user details](../users/#manage-user-settings). Once saved, the decision to make a role a "global role" can't be reverted.
```
Please let me know if you disagree.

Author: Accepted

> * **Work packages...** - tick to allow work packages to be assigned to a user with this role. This does not appear on global roles.

Reviewer: ```suggestion
3. **Work packages...** - tick to allow work packages to be assigned to a user with this role. This does not appear for global roles.
```

Author: Accepted

> * **Copy workflow from** - select an existing role. The respective [workflows](../../manage-work-packages/work-package-workflows) will be copied to this role.

Reviewer: ```suggestion
4. **Copy workflow from** - select an existing role. The respective [workflows](../../manage-work-packages/work-package-workflows) will be copied to the role to be created.
```

Author: Accepted

> You can specify the permissions per OpenProject module. Click the arrow next to the module name to expand or compress the permissions list.

Reviewer: ```suggestion
5. **Permissions** for this role - You can specify the permissions per OpenProject module. Click the arrow next to the module name to expand or compress the permissions list.
```

Author: Accepted

> 

Reviewer: ```suggestion
```

Author: Accepted
